src/hypothesis/core.py

```diff
@@ -998,7 +1038,16 @@ def wrapped_test(*arguments, **kwargs):
                 return
 
             try:
-                state.run()
+                if isinstance(cls, type) and issubclass(cls, TestCase) and \
+                        hasattr(cls, 'subTest'):
+                    subTest = cls.subTest
+                    try:
+                        setattr(cls, 'subTest', fake_subTest)
```

Comment:
A couple clarifications:

- It looks like prior to your PR, subtests "work" (correctly pass/fail when they should) but spew output for every example. (This is contrary to the impression I got from https://github.com/HypothesisWorks/hypothesis-python/issues/1071, so I guess it must have been "fixed" sometime?)

- It looks to me like after your change, we only get output for the first failing subtest. (Here's [a gist with example output](https://gist.github.com/dchudz/678687d78fb0f27aa700b41f322fe5f6).) Is that right?

I've never used `subTest`, but doing my best to think this through:

- I think a user would expect to see output for all subtests for the minimized example (which is what I see if I don't use `@given`). Is that hard to give them? (I still haven't wrapped my head around a lot of how this works, so I have no idea.)

- I think this PR probably represents an improvement in the status quo -- I'd rather just see the first subtest failure and not the spam that get the spam.

- If we aren't going to show output for all of the subtests, I think that's surprising and ideally we should signpost that pretty heavily -- possibly by issuing a warning in the test output. (The user isn't getting *any* value from using subtest, right?)

Reply:
Just to reiterate, I think this PR is an improvement without addressing any of this. Just thought I'd ask.

Reply:
Clarifications: correct on the first point; the problem there was that all the intermediate output is via `unittest`, and Hypothesis knows nothing about it (so "working as unintended", I guess).  And correct, `subTest` becomes a no-op so we only get the first failure.

Showing output for each is probably possible, but sensibly minimising for each subTest is next to impossible - they can be arbitrary code and arbitrary labels, so there's no consistent way to target them.  On this basis I've reverted to my earlier plan of putting a deprecation in the `fake_subTest` - thanks for mentioning that!